lib/shopify_api/context.rb

```diff
@@ -69,16 +72,22 @@ def setup(
         @scope = Auth::AuthScopes.new(scope)
         @is_embedded = is_embedded
         @session_storage = session_storage
-        if @session_storage
-          ::ShopifyAPI::Context.logger.warn("The use of SessionStorage in the API library has been deprecated. " \
-            "The ShopifyAPI will no longer have responsibility for session persistence. " \
-            "Upgrading to `shopify_app` 21.3 will allow you to remove session_storage" \
-            " from the API library Context configuration.")
-        end
         @logger = logger
         @private_shop = private_shop
         @user_agent_prefix = user_agent_prefix
         @old_api_secret_key = old_api_secret_key
+        @log_level = if valid_log_level?(log_level)
+          log_level
+        else
+          :info
+        end
+
+        if @session_storage
+          ::ShopifyAPI::Logger.deprecated("The use of SessionStorage in the API library has been deprecated. " \
```

Comment:
Same point I made on the[ other repo](https://github.com/Shopify/shopify_app/pull/1581#discussion_r1030843509)

Reply:
I'm a fan of iterative improvement. This is the first time we've really used deprecation logging in our api / app gem. Can we ship this as is for now and if we want to invest in more feature rich IDE tooling in the future, we can do so then?